src/ray/core_worker/task_manager.cc

```diff
@@ -272,20 +272,24 @@ void TaskManager::PendingTaskFailed(const TaskID &task_id, rpc::ErrorType error_
         << "Tried to complete task that was not pending " << task_id;
     spec = it->second.spec;
     num_retries_left = it->second.num_retries_left;
-    if (num_retries_left == 0) {
+    if (num_retries_left == -1) {
```

Comment:
These conditions would be more clear if you ordered them so the `num_retries_left > 0` and `num_retries_left == -1` cases were adjacent

Reply:
done